core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/disk/resize/DiskResizeHandler.java

```diff
@@ -0,0 +1,149 @@
+        fstabInformation.forEach((hostname, value) -> {
+            Optional<String> instanceIdOptional = stack.getInstanceMetaDataAsList().stream()
+                .filter(instanceMetaData -> hostname.equals(instanceMetaData.getDiscoveryFQDN()))
+                .map(InstanceMetaData::getInstanceId)
+                .findFirst();
+
+            if (instanceIdOptional.isPresent()) {
+                String uuids = value.getOrDefault("uuids", "");
+                String fstab = value.getOrDefault("fstab", "");
+                if (!StringUtils.isEmpty(uuids) && !StringUtils.isEmpty(fstab)) {
+                    LOGGER.debug("Persisting resources for instance id - {}, hostname - {}, uuids - {}, fstab - {}.", instanceIdOptional.get(), hostname,
+                        uuids, fstab);
+                    persistUuidAndFstab(stack, instanceIdOptional.get(), hostname, uuids, fstab);
+                }
+            }
+        });
+    }
+
+    private void persistUuidAndFstab(Stack stack, String instanceId, String discoveryFQDN, String uuids, String fstab) {
+        resourceService.saveAll(stack.getDiskResources().stream()
+            .filter(volumeSet -> instanceId.equals(volumeSet.getInstanceId()))
+            .peek(volumeSet -> resourceAttributeUtil.getTypedAttributes(volumeSet, VolumeSetAttributes.class).ifPresent(volumeSetAttributes -> {
+                volumeSetAttributes.setUuids(uuids);
+                volumeSetAttributes.setFstab(fstab);
+                if (!discoveryFQDN.equals(volumeSetAttributes.getDiscoveryFQDN())) {
+                    LOGGER.info("DiscoveryFQDN is updated for {} to {}", volumeSet.getResourceName(), discoveryFQDN);
+                }
+                volumeSetAttributes.setDiscoveryFQDN(discoveryFQDN);
+                resourceAttributeUtil.setTypedAttributes(volumeSet, volumeSetAttributes);
+            }))
+            .collect(Collectors.toList()));
+    }
+
+    @Override
+    public void accept(Event<DiskResizeHandlerRequest> diskResizeHandlerRequestEvent) {
+        LOGGER.debug("Starting resizeDisks on DiskUpdateService");
+        DiskResizeHandlerRequest payload = diskResizeHandlerRequestEvent.getData();
+        Long stackId = payload.getResourceId();
+        String instanceGroup = payload.getInstanceGroup();
+        Stack stack = stackService.getByIdWithListsInTransaction(stackId);
+        ResourceType diskResourceType = stack.getDiskResourceType();
+        if (diskResourceType != null) {
+            LOGGER.debug("Collecting resources based on stack id {} and resource type {} filtered by instance group {}.", stackId, diskResourceType,
+                    instanceGroup);
+            List<Resource> resourceList = resourceService.findAllByStackIdAndResourceTypeIn(stackId, List.of(diskResourceType))
+                    .stream().filter(res -> null != res.getInstanceId() && res.getResourceType().toString().contains("VOLUMESET")
```

Comment:
Let the database handle this filtering, to minimise the result set

Reply:
done